api/v1alpha1/vaultsecret_types.go

```diff
@@ -0,0 +1,173 @@
+	// TemplatizedK8sSecret is the formatted K8s Secret created by templating from the Vault KV secrets.
+	// +kubebuilder:validation:Required
+	TemplatizedK8sSecret TemplatizedK8sSecret `json:"output,omitempty"`
+}
+
+// VaultSecretStatus defines the observed state of VaultSecret
+type VaultSecretStatus struct {
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	//LastVaultSecretUpdate last time when this secret was updated from Vault
+	LastVaultSecretUpdate *metav1.Time `json:"lastVaultSecretUpdate,omitempty"`
+}
+
+func (vs *VaultSecret) GetConditions() []metav1.Condition {
+	return vs.Status.Conditions
+}
+
+func (vs *VaultSecret) SetConditions(conditions []metav1.Condition) {
+	vs.Status.Conditions = conditions
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// VaultSecret is the Schema for the vaultsecrets API
+type VaultSecret struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VaultSecretSpec   `json:"spec,omitempty"`
+	Status VaultSecretStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// VaultSecretList contains a list of VaultSecret
+type VaultSecretList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []VaultSecret `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&VaultSecret{}, &VaultSecretList{})
+}
+
+type KVSecret struct {
+	// Name is an arbitrary, but unique, name for this KV Vault secret and referenced when templating.
+	// +kubebuilder:validation:Required
+	Name string `json:"name,omitempty"`
+	// Authentication is the kube aoth configuraiton to be used to execute this request
+	// +kubebuilder:validation:Required
+	Authentication KubeAuthConfiguration `json:"authentication,omitempty"`
+	// Keys is a list of keys to use for templating. If none are listed all keys are referenceable for templating.
+	// +kubebuilder:validation:Optional
+	Keys []string `json:"keys,omitempty"`
+}
+
+type TemplatizedK8sSecret struct {
+	// Name is the K8s Secret name to output to.
+	// +kubebuilder:validation:Required
+	Name string `json:"name,omitempty"`
+	// Type is the K8s Secret type to output to.
+	// +kubebuilder:validation:Required
+	Type string `json:"type,omitempty"`
+	// StringData is the K8s Secret stringData and allows specifying non-binary secret data in string form with go templating support
+	// to transform the Vault KV secrets into a formatted K8s Secret.
+	// The Sprig template library and Helm functions (like toYaml) are supported.
+	// +kubebuilder:validation:Required
+	StringData map[string]string `json:"stringData,omitempty"`
+	// Annotations are annotations to add to the final K8s Secret.
+	// +kubebuilder:validation:Optional
+	Annotations map[string]string `json:"annotations,omitempty"`
```

Comment:
If annotaitons, the we also need labels...

Reply:
labels added